File: src_simple/python/produit.py
#!/usr/bin/env python
# -*- coding=utf-8 -*-

""" produit  -  04/02/2016 """


def prog_produit(n):
    """ Calcul du produit des n premiers entiers impairs """

    # initialisation
    produit = 1

    # calcul
    for i in range(3, n + 1):
        if i % 2 == 1:
            produit *= i

    # retour
    return produit


def prog_produit2(n):
    """ Calcul du produit des n premiers entiers impairs """

    # initialisation
    produit = 1

    # calcul
    for i in range(3, n + 1, 2):
        produit *= i

    # retour
    return produit

if __name__ == "__main__":
    # lecture de n
    n = int(input("entrer n : "))
    print("Calcul du produit des ", n, " premiers entiers impairs  ")
    produit = prog_produit2(n)
    print("Résultat = ", produit)

File: src_simple/f90/produit.f90
! produit  -  11.03.2014

! ******************************************
! Calcul du produit des n premiers entiers impairs
! ******************************************

program prog_produit

    implicit none

    integer :: i, n, produit

    ! lecture de n
    write(*, *) "Entrer n : "
    read(*,*) n
    write(*, *) "Calcul du produit des premiers entiers impairs entre 1 et ", n

    ! initialisation
    produit = 1

    ! boucle 
    do i = 3, n, 1
        if (mod(i, 2) == 1) then
            produit = produit * i
        end if
    end do
    write(*, *) "produit = ", produit

    ! autre boucle 
    produit = 1
    do i = 3, n, 2
        produit = produit * i
    end do
    write(*, *) "produit = ", produit

end program prog_produit

